src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala

```diff
@@ -160,13 +230,68 @@ case class GetManyDocumentsQueryBuilder(
   }
 
   private def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = {
-    postQuery[Q](client, db, s"/$db/_all_docs", ps)
+    query[Q](client, s"/$db/_all_docs", ps)
   }
 
   private def queryByIds[Q: R](ids: Seq[String], ps: Map[String, String]): Task[Q] = {
     if (ids.isEmpty)
       Res.Error("not_found", "No IDs specified").toTask
     else
-      queryByIds[String, Q](client, db, s"/$db/_all_docs", ids, ps)
+      queryByIds[String, Q](client, s"/$db/_all_docs", ids, ps)
+  }
+}
+
+object GetManyDocumentsQueryBuilder {
+
+  type MDBuilder[ID <: DocsInResult, AM <: MissingIdsInQuery, BT <: DocType] =
+  GetManyDocumentsQueryBuilder[ID, AM, BT]
+
+  case class Builder[T: R, ID <: DocsInResult, AM <: MissingIdsInQuery]
+  (builder: MDBuilder[ID, AM, AnyDocType]) {
+    def build: QueryBasic[T] =
+      new QueryBasic[T](builder.client, builder.db, builder.params, builder.ids)
   }
+
+  case class ByTypeBuilder[K: R, V: R, D: R](
+      builder: MDBuilder[IncludeDocs[D], MissingNotAllowed, ForDocType[K, V, D]])
+      (implicit tag: ClassTag[D], kw: W[K]) {
+    def build: QueryByType[K, V, D] =
+      builder.view match {
+        case Some(view) => new QueryByType[K, V, D](
+          builder.client, builder.db, view,
+          builder.typeMappings)
+      }
+  }
+
+  type BasicBuilder = Builder[CouchKeyVals[String, CouchDocRev], ExcludeDocs, MissingNotAllowed]
+  type AllowMissingBuilder = Builder[CouchKeyValsIncludesMissing[String, CouchDocRev],
+      ExcludeDocs, MissingAllowed]
+  type IncludeDocsBuilder[D] = Builder[CouchDocs[String, CouchDocRev, D], IncludeDocs[D],
+      MissingNotAllowed]
+  type AllowMissingIncludeDocsBuilder[D] = Builder[CouchDocsIncludesMissing[String, CouchDocRev,
+      D], IncludeDocs[D], MissingAllowed]
+
+  implicit def buildBasic(builder: MDBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType]):
+  BasicBuilder = new BasicBuilder(builder)
+
+  implicit def buildAllowMissing(builder: MDBuilder[ExcludeDocs, MissingAllowed, AnyDocType]):
+  AllowMissingBuilder = new AllowMissingBuilder(builder)
+
+  implicit def buildIncludeDocs[D: R](
+      builder: MDBuilder[IncludeDocs[D], MissingNotAllowed, AnyDocType]): IncludeDocsBuilder[D] =
+    new IncludeDocsBuilder(builder)
+
+  implicit def buildIncludeDocsAllowMissing[D: R](
+      builder: MDBuilder[IncludeDocs[D], MissingAllowed, AnyDocType]):
+  AllowMissingIncludeDocsBuilder[D] = new AllowMissingIncludeDocsBuilder(builder)
+
+  implicit def buildByTypeIncludeDocs[K: R, V: R, D: R](
+      builder: MDBuilder[IncludeDocs[D], MissingNotAllowed, ForDocType[K, V, D]])
+      (implicit tag: ClassTag[D], kw: W[K]): ByTypeBuilder[K, V, D] = {
+    new ByTypeBuilder[K, V, D](builder)
+  }
+
+  def apply(client: Client, db: String, typeMapping: TypeMapping):
+  MDBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType] =
+    new MDBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType](client, db, typeMapping)
```

Comment:
Are these type annotations required here? Can they be inferred based on the return type?